osf_tests/factories.py

```diff
@@ -847,36 +843,59 @@ def create_version(cls, *args, **kwargs):
             materialized_path=f'/{filename}'
         )
         preprint_file.save()
-        # TODO: why setting machine state here? it should happen in `if finish`
-        # TODO: instance.machine_state = machine_state
+        auth = Auth(instance.creator)
+        instance.set_primary_file(preprint_file, auth=auth, save=True)
         from addons.osfstorage import settings as osfstorage_settings
-        preprint_file.create_version(user, {
+        location =  {
             'object': '06d80e',
             'service': 'cloud',
             osfstorage_settings.WATERBUTLER_RESOURCE: 'osf',
-        }, {
-            'size': file_size,
+        }
+        metadata = {
+            'size': 1357,
             'contentType': 'img/png'
-        }).save()
+        }
+        preprint_file.create_version(creator, location, metadata=metadata).save()
+
+        # Set relationships
+        contributor_list = latest_version.contributor_set.exclude(
+            user=latest_version.creator
+        ).values('visible', 'user_id', '_order')
+        for contributor in contributor_list:
+            instance.contributor_set.create(**{**contributor, 'preprint_id': instance.id})
+        for institution in latest_version.affiliated_institutions.all():
+            instance.add_affiliated_institution(institution, auth.user, ignore_user_affiliation=True)
+
+        # Set other fields
+        # TODO: we should copy all fields
```

Comment:
Note: will defer this to work in https://openscience.atlassian.net/browse/ENG-6701 where this factory will be tested and further updated.